api/lib/opentelemetry.rb

```diff
@@ -35,6 +35,13 @@ def tracer_provider
     @tracer_provider ||= Trace::TracerProvider.new
   end
 
+  # @return [Trace::Tracer] delegates to the registered tracer provider and
+  #  returns a tracer with the given name and version. If called without without
+  #  arguments, it returns a tracer named 'default'.
+  def tracer(name = 'default', version = nil)
```

Comment:
Overall, I really like this, and it is pretty similar to a top-level helper in OTel Go. The only thing that feels slightly weird to me here is the different default for the `name` parameter here vs `TracerProvider#tracer`, where we use `nil` as the default.

Reply:
Agreed. Any objections to doing the same in `TracerProvider#tracer` (make the name default to "default").

Reply:
Does the spec have anything to say about this?

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#tracerprovider-operations

> `name` (required): This name must identify the instrumentation library (e.g. `io.opentelemetry.contrib.mongodb`) and _not_ the instrumented library. In case an **invalid name (null or empty string)** is specified, a working default Tracer implementation as a fallback is returned rather than returning null or throwing an exception.

Looks like the default should be `nil` or `''`.

Reply:
Ok. I guess it's not an option without a spec change. However, there is nothing forbidding the `tracer` delegate from defaulting to a name of `default`. My preference would be to keep this as-is if possible.

Reply:
After a little more thought, there's really no reason why the default tracer should be named 'default'. An unnamed tracer is as good as any. I've updated this to be consistent with the TracerProvider#tracer defaults.